### PR TITLE
Add market price (get rates) from aquanow

### PR DESCRIPTION
**Pull request by pratikjain5**
Here is get-rates function implemented and tested to get rates from Aquanow api. We also want to implement execution of cryptocurreny, could you please let us know what needs to be done for execution and how to test execution.

Tested get-rates from Tester:
./server_extensions_test/build/install/server_extensions_test/bin/server_extensions_test -a get-rates -j /batm_public/server_extensions_extra/build/libs/batm_server_extensions_extra.jar -n=aquanowapi

### CODE REVIEW

**Comment by pratikjain5** [comment]
@generalbytes @rpanak-generalbytes Hey, when can I expect this pull request to be reviewed. Could you please help us in integrating our trade execution as well with general bytes?

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aquanow/AquanowExtension.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+public class AquanowExtension extends AbstractExtension{
+    @Override
+    public String getName() {
+        return "BATM Aquanow extension";
+    }
+
+    @Override
+    public ICryptoAddressValidator createAddressValidator(String cryptoCurrency) {
+        if (CryptoCurrency.SMART.getCode().equalsIgnoreCase(cryptoCurrency)) {
```
Comment (generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aquanow/AquanowExtension.java]: SMART? Fix this.

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aquanow/AquanowExtension.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            }
+
+        }
+        return null;
+    }
+
+    @Override
+    public Set<String> getSupportedCryptoCurrencies() {
+        Set<String> result = new HashSet<String>();
+        result.add(CryptoCurrency.BTC.getCode());
```
Comment (generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aquanow/AquanowExtension.java]: MISSING AQUA?

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aquanow/sources/aquanow/AquanowRateSource.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    @Override
+    public String getPreferredFiatCurrency() {
+        return this.preferedFiatCurrency;
+    }
+
+    @Override
+    public Set<String> getCryptoCurrencies() {
+        Set<String> result = new HashSet<String>();
+        result.add(CryptoCurrency.BTC.getCode());
+        result.add(CryptoCurrency.ETH.getCode());
```
Comment (generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aquanow/sources/aquanow/AquanowRateSource.java]: AQUA. Missing?

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/sources/coinpaprika/CoinPaprikaRateSource.java [hunk 4]
```diff
@@ -76,6 +76,7 @@
         CRYPTOCURRENCIES.put(CryptoCurrency.POT.getCode(), "pot-potcoin");
         CRYPTOCURRENCIES.put(CryptoCurrency.REP.getCode(), "rep-augur");
         CRYPTOCURRENCIES.put(CryptoCurrency.SMART.getCode(), "smart-smartcash");
+        CRYPTOCURRENCIES.put(CryptoCurrency.SMART.getCode(), "smart-aquanow");
```
Comment (generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/sources/coinpaprika/CoinPaprikaRateSource.java]: Invalid. Cannot be there twice.

File: server_extensions_extra/src/main/resources/batm-extensions.xml [hunk 5]
```diff
@@ -733,6 +733,13 @@
         </ratesource>
         <cryptologo cryptocurrency="SMART" file="smart.png"/>
     </extension>
+    <extension class="com.generalbytes.batm.server.extensions.extra.aquanow.AquanowExtension" >
+        <ratesource prefix="aquanowapi" name ="Aquanow API" >
+            <param name="price" />
+            <cryptocurrency>USD is default fiat currency.</cryptocurrency>
+        </ratesource>
+        <cryptologo cryptocurrency="SMART" file="smart.png"/>
```
Comment (generalbytes) [inline server_extensions_extra/src/main/resources/batm-extensions.xml]: Duplicated

File: currencies/src/main/java/com/generalbytes/batm/common/currencies/CryptoCurrency.java [hunk 6]
```diff
@@ -77,6 +77,7 @@
     POT("PotCoin"),
     REP("Augur"),
     SMART("SmartCash"),
+    AQUANOW("Aquanow"),
```
Comment (generalbytes) [inline currencies/src/main/java/com/generalbytes/batm/common/currencies/CryptoCurrency.java]: You declare this currency, but you don't use it anywhere.

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aquanow/AquanowExtension.java (hunk 1 above)
Comment (pratikjain5) [reply server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aquanow/AquanowExtension.java]: Changed it to BTC.

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aquanow/AquanowExtension.java (hunk 2 above)
Comment (pratikjain5) [reply server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aquanow/AquanowExtension.java]: No, we don't have any cryptocurrency.

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aquanow/sources/aquanow/AquanowRateSource.java (hunk 3 above)
Comment (pratikjain5) [reply server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aquanow/sources/aquanow/AquanowRateSource.java]: No, we don't have any cryptocurrency.

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/sources/coinpaprika/CoinPaprikaRateSource.java (hunk 4 above)
Comment (pratikjain5) [reply server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/sources/coinpaprika/CoinPaprikaRateSource.java]: Removed. We don't have any cryptocurrency.

File: server_extensions_extra/src/main/resources/batm-extensions.xml (hunk 5 above)
Comment (pratikjain5) [reply server_extensions_extra/src/main/resources/batm-extensions.xml]: Removed. We don't have any cryptocurrency.

File: currencies/src/main/java/com/generalbytes/batm/common/currencies/CryptoCurrency.java (hunk 6 above)
Comment (pratikjain5) [reply currencies/src/main/java/com/generalbytes/batm/common/currencies/CryptoCurrency.java]: Removed. We don't have any cryptocurrency.

**Comment by pratikjain5** [comment]
Thanks @generalbytes for reviewing this pull request. I have made the changes requested. We don't have any crypto currency, so removed AQUA from cryptocurrency. Let me know if there is any other changes needed for integrating get-rates function.

Could you please also let us know how can we integrate execution (trading) using this repo? Do we have to create exchange in git@github.com:knowm/XChange.git repo first? So, that we can get default exchange specifications such as

org.knowm.xchange.bitfinex.v1.BitfinexExchange().getDefaultExchangeSpecification()

Or adding an exchange in knowm/XChange.git repo is not required if we just want execution(trades) on generalbytes?
Thanks, looking for your response.

**Comment by pratikjain5** [comment]
@generalbytes @rpanak-generalbytes @pvyhnal-generalbytes A gentle reminder, please let me know if there is anything else needs to be done for aquanow-extension to be added into next release. I assume it has everything for get-rates.

**Comment by pratikjain5** [comment]
@generalbytes I would really appreciate any feedback on this pull request.

**Comment by generalbytes** [comment]
Requested changes were not performed yet. Please finish your PR.

**Pull request closed without merging**